src/mobile/src/ui/views/onboarding/SetAccountName.js

```diff
@@ -108,64 +107,56 @@ export class SetAccountName extends Component {
-        const trimmedAccountName = trim(this.state.accountName);
-
-        const fetch = (accountName) => {
-            this.props.setAdditionalAccountInfo({
-                addingAdditionalAccount: true,
-                additionalAccountName: accountName,
-                usedExistingSeed: false,
-            });
-
-            this.navigateTo('loading');
-        };
+    async onDonePress() {
+        const { t, onboardingComplete, accountNames, seed, password, shouldPreventAction } = this.props;
+        const accountName = trim(this.state.accountName);
 
-        if (!isEmpty(this.state.accountName)) {
-            if (!onboardingComplete) {
-                this.props.setAccountName(trimmedAccountName);
+        if (shouldPreventAction) {
+            return this.props.generateAlert('error', t('global:pleaseWait'), t('global:pleaseWaitExplanation'));
+        }
 
-                this.navigateTo('setPassword');
-            } else {
-                if (shouldPreventAction) {
-                    return this.props.generateAlert('error', t('global:pleaseWait'), t('global:pleaseWaitExplanation'));
-                }
-                getAllSeedsFromKeychain(password)
-                    .then((seedInfo) => {
-                        if (isEmpty(seedInfo)) {
-                            return fetch(trimmedAccountName);
-                        }
-                        if (hasDuplicateAccountName(seedInfo, trimmedAccountName)) {
-                            return this.props.generateAlert(
-                                'error',
-                                t('addAdditionalSeed:nameInUse'),
-                                t('addAdditionalSeed:nameInUseExplanation'),
-                            );
-                        } else if (hasDuplicateSeed(seedInfo, seed)) {
-                            return this.props.generateAlert(
-                                'error',
-                                t('addAdditionalSeed:seedInUse'),
-                                t('addAdditionalSeed:seedInUseExplanation'),
-                            );
-                        }
-                        return fetch(trimmedAccountName);
-                    })
-                    .catch(() => {
-                        this.props.generateAlert(
-                            'error',
-                            t('global:somethingWentWrong'),
-                            t('global:somethingWentWrongExplanation'),
-                        );
-                    });
-            }
-        } else {
-            this.props.generateAlert(
+        if (isEmpty(accountName)) {
+            return this.props.generateAlert(
                 'error',
                 t('addAdditionalSeed:noNickname'),
                 t('addAdditionalSeed:noNicknameExplanation'),
             );
         }
+
+        if (accountNames.map((item) => item.toLowerCase()).indexOf(accountName.toLowerCase()) > -1) {
```

Comment:
A single `seedStore` contains account info only about it's seedStore type (on desktop they are also hashed), so it would require to loop all account seedStore types to do this check.
The state should always contain the correct account name list.